### DISCUSSION TITLE
Storing intermediate values in a label

### DISCUSSION THREAD

**Discussion by DolceTriade** [HASwitchPlate/openHASP#337] [Q&A] [answered]
Hello,

I've gotten openHASP working on a wt32-sc01 and a lillygo pi. However, I would also like to control my alarm panel via openHasp, but I can't seem to figure out how to enter my unlock code via this...

There doesn't seem to be a way to store interim values that I can use to build the total code out. Do I need to make a psuedo sensor in HomeAssistant to store this and write it back to a label?

I saw that LVGL has demos with a keyboard, which I theoretically could use, but I can't seem to find the component.

Thanks!

**Comment by nagyrobi** [comment, marked as answer]
Look for tag property.

**Reply by DolceTriade** [reply]
Is there an example of how to use the tag property? I couldn't find any in the docs.

Thanks!

**Reply by nagyrobi** [reply]
https://openhasp.haswitchplate.com/0.6.3/integrations/home-assistant/sampl_conf/#common-properties

**Reply by DolceTriade** [reply]
Ohh! This is very cool. I didn't realize I could see documentation for the new version! Thanks a lot

**Reply by DolceTriade** [reply]
For posterity, this is what I ended up with:

pages.jsonl:
```
{"page":5,"id":21,"obj":"label","x":50,"y":50,"w":220,"h":30,"text":"1234","align":"center","bg_color":"#2C3E50", "text_color":"#FFFFFF"}
{"page":5,"id":20,"obj":"btnmatrix","x":50,"y":80,"w":220,"h":150,"options":["1", "2", "3", "\n", "4", "5", "6", "\n", "7", "8", "9", "\n", "\uE156", "0", "\uE12C"]}
```
Home Assistant:
```
  - obj: "p5b20"
    event:
      "down":
        - service: openhasp.command
          target:
            entity_id: openhasp.plate
          data:
            keyword: >
                  {% if val != 9 and val != 11 %}
                  p5b20.tag="{{ tag }}{{ text }}"
                  {% else %}
                  p5b20.tag=""
                  {% endif %}

        - service: openhasp.command
          target:
            entity_id: openhasp.plate
          data:
            keyword: >
                  {% if val != 9 and val != 11 %}
                  p5b21.text={{ tag }}{{ text }}
                  {% else %}
                  p5b21.text=
                  {% endif %}
```

The MQTT round trip is kinda depressing as it leads to lag, but I can't think of anything better.

**Comment by fvanroie** [comment]
> I saw that LVGL has demos with a keyboard, which I theoretically could use, but I can't seem to find the component.

Keyboard or Textarea is not implemented in openHASP (yet)...
LVGL has many, many APIs to manipulate the objects and in openHASP there is a subset available for use.
